engine.fhir/src/main/java/org/opencds/cqf/cql/engine/fhir/retrieve/RestFhirRetrieveProvider.java

```diff
@@ -28,13 +29,19 @@
 	protected IGenericClient fhirClient;
 	private SearchStyleEnum searchStyle;
 
-	public RestFhirRetrieveProvider(SearchParameterResolver searchParameterResolver, IGenericClient fhirClient) {
-		super(searchParameterResolver);
-		// TODO: Figure out how to validate that the searchParameterResolver and the
-		// client are on the same version of FHIR.
-		this.fhirClient = fhirClient;
-		this.searchStyle = DEFAULT_SEARCH_STYLE;
-	}
+    public RestFhirRetrieveProvider(SearchParameterResolver searchParameterResolver, IGenericClient fhirClient) {
+        super(searchParameterResolver);
+        this.fhirClient = fhirClient;
+        this.searchStyle = DEFAULT_SEARCH_STYLE;
+    }
+
+    public RestFhirRetrieveProvider(SearchParameterResolver searchParameterResolver, FhirModelResolver modelResolver, IGenericClient fhirClient) {
```

Comment:
Same issue here. You can't count on any interface more specific than `ModelResolver`